databricks_cli/stack/api.py

```diff
@@ -139,6 +162,84 @@ def deploy_config(self, stack_config, stack_status=None, **kwargs):
 
         return new_stack_status
 
+    def download_config(self, stack_config, **kwargs):
+        """
+        Downloads a stack given a dict of the stack configuration
+        :param stack_config:
+        :param kwargs:
+        :return:
+        """
+        self._validate_config(stack_config)
+        click.echo('#' * 80)
+        for resource_config in stack_config.get(STACK_RESOURCES):
+            # Deploy resource, get resource_status
+            self._download_resource(resource_config, **kwargs)
+            click.echo('#' * 80)
+
+    def _download_resource(self, resource_config, **kwargs):
+        """
+        Downloads a resource given a resource information extracted from the stack JSON
+        configuration template.
+
+        :param resource_config: A dict of the resource with fields of RESOURCE_ID, RESOURCE_SERVICE
+        and RESOURCE_PROPERTIES.
+        ex. {'id': 'example-resource', 'service': 'jobs', 'properties': {...}}
+        """
+        resource_id = resource_config.get(RESOURCE_ID)
+        resource_service = resource_config.get(RESOURCE_SERVICE)
+        resource_properties = resource_config.get(RESOURCE_PROPERTIES)
+
+        if resource_service == WORKSPACE_SERVICE:
+            click.echo(
+                "Downloading workspace asset '{}' with properties \n{}"
+                .format(
+                    resource_id, json.dumps(resource_properties, indent=2, separators=(',', ': '))
+                )
+            )
+            overwrite = kwargs.get('overwrite_notebooks', False)
+            self._download_workspace(resource_properties, overwrite)
+        else:
+            click.echo("Resource service '{}' not supported for download. "
+                       "skipping.".format(resource_service))
+
+    def _download_workspace(self, resource_properties, overwrite):
+        """
+        Deploy workspace asset.
```

Comment:
"Deploy" -> "Download"?

Reply:
Yes. Actually haven't worked too much on the documentation yet, just on the code. I'll get the documentation down hopefully by monday.